controllers/storagecluster/provider_server.go

```diff
@@ -252,6 +252,10 @@ func (o *ocsProviderServer) ensureDeploymentReplica(actual, desired *appsv1.Depl
 
 func GetProviderAPIServerDeployment(instance *ocsv1.StorageCluster) *appsv1.Deployment {
 
+	// We need to copy the values from getDaemonResources into a varirable as the Pod spec expects
+	// a pointer to the ResourceRequirements api.
+	ocsProviderServerResource := getDaemonResources(ocsProviderServerName, instance)
```

Comment:
Instead of specifying the resources at Pod level try specifying them at the container level. That's the recommended way by k8s, and it does not require a pointer as well.